packages/core/src/browser/common-frontend-contribution.ts

```diff
@@ -837,6 +931,23 @@ export class CommonFrontendContribution implements FrontendApplicationContributi
         return environment.electron.is();
     }
 
+    private isPinned(title: Title<Widget>): boolean {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        return (title as any).pinned;
```

Comment:
I'd prefer a real type to this `as any` cast, something like
```typescript
interface PinnableTitle extends (Phosphor)Title { pinned?: boolean }
```
or (probably in `widget.ts` or somewhere)
```typescript
interface TheiaTitle extends Title {
    pinned?: boolean;
    // ... who knows what else in the future?
}
```
That way that information could be used elsewhere, for example if we want to override the rendering of tabbars [here](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/browser/shell/tab-bars.ts#L133-L158).